bee-common/src/shutdown.rs

```diff
@@ -26,13 +28,14 @@ pub enum Error {
 
 pub type ShutdownNotifier = oneshot::Sender<()>;
 pub type ShutdownListener = oneshot::Receiver<()>;
-pub type WorkerHandle = task::JoinHandle<Result<(), WorkerError>>;
+pub type Worker = Box<dyn Future<Output = Result<(), WorkerError>> + Unpin>;
```

Comment:
Isn't this specifically the completion future of a worker? I'd personally be more tempted to call it `WorkerShutdown` or something like that. This is also confusing since `WorkerHandle` refers to the worker itself, but `Worker` only refers to the completion future.

Reply:
I think `WorkerShutdown` works for me :slightly_smiling_face: